src/lexer/preprocessor/preprocessor.rs

```diff
@@ -166,12 +167,24 @@ impl<'a, PC: PreprocContext> Lexer<'a, PC> {
                 Token::PreprocPragma
             }
             Token::PreprocError => {
-                skip_until!(self, b'\n');
-                self.buf.add_new_line();
-                Token::PreprocError
+                let mut buf = Vec::new();
```

Comment:
Unrelated change so please do it an other commit.

Reply:
This new code is mainly there to get the error message of the `#error something` pragma. How should I put it into a separate commit? Should I first add the `#error` error without reading the message and then add the message later, or should I make one separate commit about the `#error` error?

Reply:
Ok let's do it here.
Here I'd save the pos: `let spos = self.buf.pos();`
and then skip_until '\n'.
Finally get the slice with `self.buf.slice(spos)` and then convert it into a string.